src/sentry/static/sentry/app/components/events/eventRelatedEvents.tsx

```diff
@@ -0,0 +1,152 @@
+import React from 'react';
+import {Location} from 'history';
+import uniqBy from 'lodash/uniqBy';
+
+import {t} from 'app/locale';
+import {Organization, GlobalSelection, Event} from 'app/types';
+import EventView from 'app/utils/discover/eventView';
+import DiscoverQuery from 'app/utils/discover/discoverQuery';
+import withGlobalSelection from 'app/utils/withGlobalSelection';
+import LoadingIndicator from 'app/components/loadingIndicator';
+import EventDataSection from 'app/components/events/eventDataSection';
+
+import RelatedEvents from './relatedEvents';
+import EmptyState from './relatedEvents/emptyState';
+import {getCurrentLocation} from './relatedEvents/utils';
+import DiscoverButton from './relatedEvents/discoverButton';
+
+type Props = {
+  location: Location;
+  event: Event;
+  organization: Organization;
+  selection: GlobalSelection;
+};
+
+type State = {
+  isLoading: boolean;
+  orgFeatures: Set<string>;
+  orgSlug: string;
+  period: GlobalSelection['datetime']['period'];
+  eventView?: EventView;
+};
+
+class EventRelatedEvents extends React.Component<Props, State> {
+  state: State = {
+    isLoading: true,
+    orgFeatures: new Set(this.props.organization.slug),
+    orgSlug: this.props.organization.slug,
+    period: this.props.selection.datetime.period,
+  };
+
+  componentDidMount() {
+    this.getEventView();
+  }
+
+  getEventView() {
+    const {event, organization} = this.props;
+
+    // traceId should always be defined, expect if the customers are using old SDK's, or SDKs on native where there isn't any tracing support
+    const traceId = event.contexts?.trace?.trace_id;
+
+    if (!traceId) {
+      this.setState({isLoading: false});
+      return;
+    }
+
+    const {period} = this.state;
+    const orgFeatures = new Set(organization.features);
+
+    const eventFromSavedQuery = EventView.fromSavedQuery({
+      id: undefined,
+      name: `Events with Trace ID ${traceId}`,
+      fields: [
+        'title',
+        'event.type',
+        'project',
+        'project.id',
+        'trace.span',
+        'timestamp',
+        'lastSeen',
+        'issue',
+      ],
+      orderby: '-timestamp',
+      query: `trace:${traceId}`,
+      projects: orgFeatures.has('global-views') ? [-1] : [Number(event.projectID)],
```

Comment:
You can use `ALL_ACCESS_PROJECTS` constant from `app/constants/globalSelectionHeader`

I recently applied this here for the span view https://github.com/getsentry/sentry/pull/20161